File: src/main/java/org/wcardinal/controller/data/internal/SMovableListPatch.java
/*
 * Copyright (C) 2019 Toshiba Corporation
 * SPDX-License-Identifier: Apache-2.0
 */

package org.wcardinal.controller.data.internal;

import java.util.List;
import java.util.NavigableMap;

import org.wcardinal.controller.data.SMovableList.Move;
import org.wcardinal.controller.data.SList.Update;

public interface SMovableListPatch<V> extends SListPatch<V> {
	void move( int oldIndex, int newIndex );
	void apply( final List<V> list, final NavigableMap<Integer, V> padded, final NavigableMap<Integer, V> premoved,
		final NavigableMap<Integer, Update<V>> pupdated, final List<Move<V>> pnewMoved );
}

File: src/main/typescript/wcardinal/controller/data/internal/s-movable-list-patch.ts
/*
 * Copyright (C) 2019 Toshiba Corporation
 * SPDX-License-Identifier: Apache-2.0
 */

import { SListBasePatch } from "./s-list-base-patch";

export interface SMovableListPatch<V> extends SListBasePatch<V> {
	move_( oldIndex: number, newIndex: number ): void;
}
